app/uk/gov/hmrc/cgtpropertydisposalsfrontend/views/returns/task_list.scala.html

```diff
@@ -34,6 +39,39 @@
 @propertyAddressState = @{ draftReturn.propertyAddress.fold[TaskListStatus](TaskListStatus.ToDo)(_ => TaskListStatus.Complete) }
 @disposalDetailsState = @{ draftReturn.disposalDetailsAnswers.fold[TaskListStatus](TaskListStatus.ToDo)(_.fold(_ => TaskListStatus.InProgress,_ => TaskListStatus.Complete)) }
 @acquistionDetailsState = @{ draftReturn.acquisitionDetailsAnswers.fold[TaskListStatus](TaskListStatus.ToDo)(_.fold(_ => TaskListStatus.InProgress,_ => TaskListStatus.Complete)) }
+
+@initialGainOrLossState = @{
+    def canUseService = {
+        val residentialBeforeDate = LocalDateUtils.dateFromString("1/04/2015")
+        val nonResidentialBeforeDate = LocalDateUtils.dateFromString("1/04/2019")
+        (triageState) match {
+            case (TaskListStatus.Complete) => {
+                val CompleteSingleDisposalTriageAnswers(_, _, countryOfResidence, assetType, _, _) = draftReturn.triageAnswers
+
+                (countryOfResidence, assetType, draftReturn.acquisitionDetailsAnswers) match {
+                    case (_, _, _) if countryOfResidence.isUk() => false
+                    case (_, AssetType.NonResidential, Some(CompleteAcquisitionDetailsAnswers(_, AcquisitionDate(acquisitionDate), _, _, _, _)))
+                        if (acquisitionDate.isBefore(nonResidentialBeforeDate)) => true
+                    case (_, AssetType.Residential, Some(CompleteAcquisitionDetailsAnswers(_, AcquisitionDate(acquisitionDate), _, _, _, _)))
+                        if(acquisitionDate.isBefore(residentialBeforeDate)) => true
+                    case (_, _, _) => false
+
+                }
+            }
+            case _ => false
+        }
+    }
+
+    if (canUseService)
+        draftReturn.initialGainOrLossAnswers match {
+         case(Some(_)) => TaskListStatus.Complete
+            case _ => TaskListStatus.ToDo
+        }
+     else
```

Comment:
shouldn't it be that this section is completely hidden from the task list in some scenarios?

Reply:
Done.